src/main/scala/PropertyChecker.scala

```diff
@@ -630,4 +664,285 @@ object PropertyChecker {
+        case a: ArrayNode => (a, Array[String](), csvwPropertyType)
+        case _ => (BooleanNode.getFalse, Array[String](PropertyChecker.invalidValueWarning), csvwPropertyType)
+      }
+    }
+  }
+
+  def trimProperty(csvwPropertyType: PropertyType.Value):(JsonNode, String, String) => (JsonNode, Array[String], PropertyType.Value) = {
+    (value, baseUrl, lang) => {
+      var valueCopy: JsonNode = value.deepCopy()
+      valueCopy match {
+        case b: BooleanNode => {
+          if (b.booleanValue) {
+            valueCopy = new TextNode("true")
+          } else {
+            valueCopy = new TextNode("false")
+          }
+        }
+        case _ => {}
+      }
+      if (Array[String]("true", "false", "start", "end").contains(valueCopy.asText())) {
+        (valueCopy, Array[String](), PropertyType.Dialect)
+      } else {
+        (BooleanNode.getTrue, Array[String](PropertyChecker.invalidValueWarning), PropertyType.Dialect)
+      }
+    }
+  }
+
+  def columnsProperty(csvwPropertyType: PropertyType.Value):(JsonNode, String, String) => (JsonNode, Array[String], PropertyType.Value) = {
+    (value, baseUrl, lang) => {
+      (value, Array[String](), csvwPropertyType)
+    }
+  }
+
+  def columnReferenceProperty(csvwPropertyType: PropertyType.Value):(JsonNode, String, String) => (JsonNode, Array[String], PropertyType.Value) = {
+    (value, baseUrl, lang) => {
+      value match {
+        case s: TextNode => {
+          val arrayNode = JsonNodeFactory.instance.arrayNode()
+          arrayNode.add(s)
+          (arrayNode, Array[String](), csvwPropertyType)
+        }
+        case a: ArrayNode => (a, Array[String](), csvwPropertyType)
+      }
+    }
+  }
+
+  def targetFormatProperty(csvwPropertyType: PropertyType.Value):(JsonNode, String, String) => (JsonNode, Array[String], PropertyType.Value) = {
+    (value, baseUrl, lang) => {
+      (value, Array[String](), csvwPropertyType)
+    }
+  }
+
+  def scriptFormatProperty(csvwPropertyType: PropertyType.Value):(JsonNode, String, String) => (JsonNode, Array[String], PropertyType.Value) = {
+    (value, baseUrl, lang) => {
+      (value, Array[String](), csvwPropertyType)
+    }
+  }
+
+  def sourceProperty(csvwPropertyType: PropertyType.Value):(JsonNode, String, String) => (JsonNode, Array[String], PropertyType.Value) = {
+    (value, baseUrl, lang) => {
+      (value, Array[String](), csvwPropertyType)
+    }
+  }
+
+  def resourceProperty(csvwPropertyType: PropertyType.Value):(JsonNode, String, String) => (JsonNode, Array[String], PropertyType.Value) = {
+    (value, baseUrl, lang) => {
+      (value, Array[String](), csvwPropertyType)
+    }
+  }
+
+  def schemaReferenceProperty(csvwPropertyType: PropertyType.Value):(JsonNode, String, String) => (JsonNode, Array[String], PropertyType.Value) = {
+    (value, baseUrl, lang) => {
+      val url = baseUrl + value.asText()
+      // Don't know how to place a URI object in JsonNode, keeping the text value as of now
```

Comment:
URIs are represented as text in JSON, so that's perfectly fine.

I'm surprised there's no type-checking of the nodes in `targetFormatProperty`, `sourceProperty`, `resourceProperty` and here. Any idea why?

Reply:
Those are mainly empty lambdas. It just forwards the property as received. No validation logic is present for these properties

Reply:
I take it that's what's happening in the ruby implementation. I'm just wondering whether we should be doing more validation.

Reply:
I think in our last discussion on this, we said will trust the ruby property checking for now and will come back to it something is not right